core/frontend/src/components/kraken/ExtensionCreationDialog.vue

```diff
@@ -0,0 +1,182 @@
+            v-if="new_extension.permissions"
+            v-model="new_extension.permissions"
+            label="Original Permissions"
+            :readonly="true"
+            :rules="[validate_permissions]"
+          />
+
+          <v-textarea
+            v-model="new_extension.user_permissions"
+            label="Custom Permissions (These replace regular permissions)"
+            :rules="[validate_permissions]"
+          />
+
+          <v-btn
+            color="primary"
+            class="mr-4"
+            @click="saveExtension"
+          >
+            Create
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue'
+
+import Notifier from '@/libs/notifier'
+import { kraken_service } from '@/types/frontend_services'
+import { InstalledExtensionData } from '@/types/kraken'
+import { VForm } from '@/types/vuetify'
+import back_axios from '@/utils/api'
+
+const notifier = new Notifier(kraken_service)
+
+export default Vue.extend({
+  name: 'ExtensionCreationDialog',
+  model: {
+    prop: 'extension',
+    event: 'change',
+  },
+  props: {
+    extension: {
+      type: Object as PropType<InstalledExtensionData | null>,
+      default: null,
+    },
+  },
+
+  data() {
+    return {
+      new_extension: this.extension,
+    }
+  },
+  computed: {
+    form(): VForm {
+      return this.$refs.form as VForm
+    },
+  },
+  watch: {
+    new_extension() {
+      this.$emit('input', this.new_extension)
+    },
+    extension() {
+      this.new_extension = this.extension
+    },
+  },
+  methods: {
+    closeDialog() {
+      this.new_extension = null
+    },
+    validate_identifier(input: string): (true | string) {
+      // Identifiers should be two words separated by a period
+      // They can contain lower and uppercase characters, but cannot begin with a number
+      const re = /^[A-Za-z][A-Za-z0-9]*\.[A-Za-z]+[A-Za-z0-9]*/
+      if (re.test(input)) {
+        return true
+      }
+      return 'This field must container two words separated by a period. Numbers are allowed if not the first character'
+    },
+    validate_dockerhub(input: string): (true | string) {
+      // A tag name must be valid ASCII and may contain lowercase and uppercase letters, digits,
+      // underscores, periods and dashes. A tag name may not start with a period or a dash and
+      // may contain a maximum of 128 characters.
+      // regex based on https://ktomk.github.io/pipelines/doc/DOCKER-NAME-TAG.html#grammar
+      const re = /[A-Za-z0-9_][A-Za-z0-9_.-]{0,127}/
+      if (re.test(input)) {
+        return true
+      }
+      return 'This field must container two words separated by a forward slash'
+      + 'Numbers are allowed if not the first character. e.g example/docker1'
+    },
+    validate_name(input: string): (true | string) {
+      // Lowercase, uppercase characters, and spaces
+      const re = /[A-Za-z0-9 ]/
```

Comment:
This regex only validate the first character, ' {}{}{}{}' is valid since the first character is a space.